CMakeLists.txt

```diff
@@ -3,9 +3,9 @@
 #-----------------------------------------------------------------------
 # - Enforce an out-of-source builds before anything else
 #
-if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+if(${Geant4_SOURCE_DIR} STREQUAL ${Geant4_BINARY_DIR})
   message(STATUS "Geant4 requires an out-of-source build.")
-  message(STATUS "Please remove these files from ${CMAKE_BINARY_DIR} first:")
+  message(STATUS "Please remove these files from ${Geant4_BINARY_DIR} first:")
```

Comment:
These probably need to be `CMAKE_CURRENT_{SOURCE,BINARY}_DIR`. Since this happens _before_ invocation of the `project` command, the `Geant4_...` variables will be empty the first time CMake runs.